tools/toolchains/arm.py

```diff
@@ -56,6 +56,11 @@ def __init__(self, target, options=None, notify=None, macros=None, silent=False,
         else:
             cpu = target.core
 
+        if not TOOLCHAIN_PATHS['ARM']:
+            exe = find_executable('armcc')
+            if exe:
```

Comment:
ignore my comment. I can see that TOOLCHAIN_PATHS['ARM'] is already initialized tools.toolchains.__init__.py